packages/react/src/components/Authenticator/Authenticator.tsx

```diff
@@ -1,86 +1,128 @@
     if (!hasInitialized.current && route === 'setup') {
-      _send({
-        type: 'INIT',
-        data,
-      });
+      _send({ type: 'INIT', data });
+
       hasInitialized.current = true;
     }
   }, [_send, route, data]);
-  return <>{children}</>;
+
+  const isAuthenticatedRoute = route === 'authenticated' || route === 'signOut';
+  const ChildComponent = React.useMemo(() => {
+    if (!isAuthenticatedRoute) {
+      return children;
+    }
+
+    // `Authenticator` might not have user defined `authenticatorChildren` for non SPA use cases.
+    if (!authenticatorChildren) {
+      return null;
+    }
+
+    return typeof authenticatorChildren === 'function'
+      ? authenticatorChildren({ signOut, user }) // authenticatorChildren is a render prop
+      : authenticatorChildren;
+  }, [authenticatorChildren, children, isAuthenticatedRoute, signOut, user]);
+
+  return ChildComponent;
 }
 
-export function Authenticator({
-  children,
+// use Authenticator namespace for both the component and the interface
+// to ensure that Typescript adds the correct properties on the component
+interface Authenticator {
+  (props: AuthenticatorProps): JSX.Element;
+  Provider?: (children: React.ReactNode) => JSX.Element;
+  ForceNewPassword?: DefaultComponents['ForceNewPassword'];
+  ResetPassword?: DefaultComponents['ResetPassword'];
+  SetupTOTP?: DefaultComponents['SetupTOTP'];
+  SignIn?: DefaultComponents['SignIn'];
+  SignUp?: DefaultComponents['SignUp'];
+}
+
+export const Authenticator: Authenticator = ({
+  children: authenticatorChildren,
   className,
   components: customComponents,
+  formFields,
+  hideSignUp,
   initialState,
   loginMechanisms,
-  services,
   signUpAttributes,
+  services,
   socialProviders,
   variation,
-  hideSignUp,
-  formFields,
-}: AuthenticatorProps) {
+}) => {
   const components = { ...defaultComponents, ...customComponents };
   const machineProps = {
+    formFields,
     initialState,
     loginMechanisms,
     services,
     signUpAttributes,
     socialProviders,
-    formFields,
   };
 
   return (
     <Provider>
       <CustomComponentsContext.Provider value={{ components }}>
-        <InitMachine {...machineProps}>
-          <Router
-            className={className}
-            children={children}
-            variation={variation}
-            hideSignUp={hideSignUp}
-          />
+        <InitMachine
+          {...machineProps}
+          authenticatorChildren={authenticatorChildren}
+        >
+          <AuthenticatorContainer className={className} variation={variation}>
+            <Router hideSignUp={hideSignUp} />
+          </AuthenticatorContainer>
         </InitMachine>
       </CustomComponentsContext.Provider>
     </Provider>
   );
-}
-
-Authenticator.Provider = Provider;
-Authenticator.ResetPassword = ResetPassword;
-Authenticator.SetupTOTP = SetupTOTP;
-Authenticator.SignIn = SignIn;
-Authenticator.SignUp = SignUp;
-Authenticator.ForceNewPassword = ForceNewPassword;
```

Comment:
Moved the assigning of these components as properties to the primary _index.ts_ that the `Authenticator` is exported from. Something abt the `AuthenticatorContainer` being imported in to this file and wrapping the `Router` along with these property assignments was causing the bundle to get corrupted.